### PR TITLE
EHUB-272 Add private catalog api key management

### PR DESCRIPTION
**Pull request by alepintus**

### CODE REVIEW

File: application/microservices/entando-hub-catalog-ms/pom.xml [hunk 1]
```diff
@@ -192,6 +194,33 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>9</source>
+                    <target>9</target>
```
Comment (firegloves) [inline application/microservices/entando-hub-catalog-ms/pom.xml]: Is version 9 a requirement or can we switch to 11?

File: application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/rest/PrivateCatalogApiKeyController.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    @ApiResponse(responseCode = "200", description = "OK", content = @Content)
+    @ApiResponse(responseCode = "400", description = "Bad Request", content = @Content)
+    @ApiResponse(responseCode = "401", description = "Unauthorized", content = @Content)
+    @ApiResponse(responseCode = "403", description = "Forbidden", content = @Content)
+    public ResponseEntity<Map<String, Boolean>> editMyLabel(@PathVariable @NotNull Long id, @Valid @RequestBody EditApiKeyRequestDTO request) {
+        String username = this.securityHelperService.getContextAuthenticationUsername();
+        boolean result = this.privateCatalogApiKeyService.editLabel(id, username, request.getLabel());
+        Map<String, Boolean> mapResult = new HashMap<>();
+        mapResult.put("result", result);
+        return new ResponseEntity<>(mapResult, HttpStatus.OK);
```
Comment (firegloves) [inline application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/rest/PrivateCatalogApiKeyController.java]: Could we return an empty response with status 200? I think the boolean is not adding any worth

File: application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/rest/PrivateCatalogApiKeyController.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    @ApiResponse(responseCode = "200", description = "OK", content = @Content)
+    @ApiResponse(responseCode = "400", description = "Bad Request", content = @Content)
+    @ApiResponse(responseCode = "401", description = "Unauthorized", content = @Content)
+    @ApiResponse(responseCode = "403", description = "Forbidden", content = @Content)
+    public ResponseEntity<Map<String, Boolean>> deleteMyApiKey(@PathVariable @NotNull Long id) {
+        String username = this.securityHelperService.getContextAuthenticationUsername();
+        boolean result = this.privateCatalogApiKeyService.deleteApiKey(id, username);
+        Map<String, Boolean> mapResult = new HashMap<>();
+        mapResult.put("result", result);
+        return new ResponseEntity<>(mapResult, HttpStatus.OK);
```
Comment (firegloves) [inline application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/rest/PrivateCatalogApiKeyController.java]: Same concern of the previous comment

File: application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/rest/handlers/RestExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+@ControllerAdvice
+public class RestExceptionHandler extends ResponseEntityExceptionHandler {
+    @ExceptionHandler({BadRequestException.class})
+    public ResponseEntity<ErrorResponse> customHandleBadRequest(Exception ex) {
+        ErrorResponse errors = new ErrorResponse();
+        errors.setTimestamp(LocalDateTime.now());
+        errors.setError(ex.getMessage());
+        errors.setStatus(HttpStatus.BAD_REQUEST.value());
+        return new ResponseEntity<>(errors, HttpStatus.BAD_REQUEST);
+    }
```
Comment (firegloves) [inline application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/rest/handlers/RestExceptionHandler.java:29]: Is this changing the management behaviour for every caught BadRequestException? In this case we should firstly check that the FE is not impacted

File: application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/service/mapper/PrivateCatalogApiKeyMapper.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    public abstract ApiKeyResponseDTO toRefreshApiKeyDto(PrivateCatalogApiKey privateCatalogApiKey);
+
+    @Named( "toApiKeyResponseDTOList" )
+    @Mapping(target = "apiKey", ignore = true)
+    public abstract List<ApiKeyResponseDTO> toApiKeyResponseDTO(List<PrivateCatalogApiKey> privateCatalogApiKey);
+
+
+    @Named( "removeApiKey" )
+    public List<ApiKeyResponseDTO> removeApiKey(List<ApiKeyResponseDTO> privateCatalogApiKey){
+        privateCatalogApiKey.forEach(f-> {System.out.println("f -> "+f.toString());f.setApiKey(null);});
```
Comment (firegloves) [inline application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/service/mapper/PrivateCatalogApiKeyMapper.java]: Is this System.out required?

File: application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/entity/PrivateCatalogApiKeyEntityTest.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
+package com.entando.hub.catalog.entity;
+
+import com.entando.hub.catalog.persistence.entity.PrivateCatalogApiKey;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import static com.entando.hub.catalog.service.PrivateCatalogApiKeyGeneratorHelper.generatePrivateCatalogApiKeyEntity;
+
+@SpringBootTest
```
Comment (firegloves) [inline application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/entity/PrivateCatalogApiKeyEntityTest.java]: Is it required by the test? It seems to me that it is not involving any spring component

**Review by firegloves — CHANGES_REQUESTED** [review]

File: application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/rest/handlers/RestExceptionHandler.java (hunk 4 above)
Comment (alepintus) [reply application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/rest/handlers/RestExceptionHandler.java:29]: I don't think, this is a new Exception class that I added in this PR and it's only used by the new api key controllers

File: application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/rest/PrivateCatalogApiKeyController.java (hunk 2 above)
Comment (alepintus) [reply application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/rest/PrivateCatalogApiKeyController.java]: done

File: application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/rest/PrivateCatalogApiKeyController.java (hunk 3 above)
Comment (alepintus) [reply application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/rest/PrivateCatalogApiKeyController.java]: done

File: application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/service/mapper/PrivateCatalogApiKeyMapper.java (hunk 5 above)
Comment (alepintus) [reply application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/service/mapper/PrivateCatalogApiKeyMapper.java]: no, logs removed

File: application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/entity/PrivateCatalogApiKeyEntityTest.java (hunk 6 above)
Comment (alepintus) [reply application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/entity/PrivateCatalogApiKeyEntityTest.java]: Annotation removed

**Review by firegloves — APPROVED** [review]
LGTM

**Pull request merged**